### PR TITLE
Feat/tdt rendering

### PR DESCRIPTION
**Pull request by skokenes**
## Checklist
- [x] Manual verification
- [ ] Unit test coverage
- [ ] E2E test coverage
- [ ] Needs manual QA?

#### Issue addressed: 
Incremental PR adding more rendering capabilities to match time dimension table spec

#### Details:
Adds
- styling column headers and column header components
- styling cells: numbers, percentages
- interactive circles in dimension cells for toggling visibility of the dimension value in the chart (TBD)

Changes
- replaced the <table> virtualization system with a <div> based virtualization system. This greatly improved performance and matches what other data grid virtualization tools use (mui datagrid, ag-grid)

https://github.com/rilldata/rill/assets/5554373/ccd1852e-5684-4bb6-94e1-73f614837eb0

## Steps to Verify
Go to `localhost:3000/dev/table/a`

### CODE REVIEW

File: web-common/src/features/dashboards/time-dimension-details/context.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { Writable } from "svelte/store";
+import { getContext, setContext } from "svelte";
+import type { TimeDimensionDetailsStore } from "./time-dimension-details-store";
+import type { THeaderData } from "./mock-data";
+
+const CONTEXT_KEY = "tdt-store";
+
+type TDTContext = {
```
Comment (ericpgreen2) [inline web-common/src/features/dashboards/time-dimension-details/context.ts:8]: Nit: Make the terminology/acronym consistent throughout? `time-details` / TDT (Time DeTails), or `time-dimension-details` / TDD (Time Dimension Details)?

**Review by ericpgreen2 — APPROVED** [review]
Looks good 👍

File: web-common/src/features/dashboards/time-dimension-details/context.ts (hunk 1 above)
Comment (skokenes) [reply web-common/src/features/dashboards/time-dimension-details/context.ts:8]: Good point, lets go with TDD. I think I started with that

File: web-common/src/features/dashboards/time-dimension-details/DimensionCell.svelte [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  import {
+    getVisibleDimensionColor,
+    toggleVisibleDimensions,
+  } from "./time-dimension-details-store";
+  import { useTDTContext } from "./context";
+  import type { TCellData } from "./mock-data";
+
+  export let cell: TCellData;
+
+  let _class = "w-full h-full flex items-center justify-between ";
```
[the remark this answers is not in the document]
Comment (skokenes) [reply web-common/src/features/dashboards/time-dimension-details/DimensionCell.svelte:11]: Yea I know what you mean. For now, we can probably get away with just using an array of strings and then joining them with spaces at the end.
The big advantage of TailwindMerge is that it handles conflicts, as far as I understand it. Would def make life easier in some places

File: web-common/src/features/dashboards/time-dimension-details/context.ts (hunk 1 above)
Comment (skokenes) [reply web-common/src/features/dashboards/time-dimension-details/context.ts:8]: FYI I am going to fix this in the next PR

**Pull request merged**